src/TSTransformer/nodes/class/transformClassLikeDeclaration.ts

```diff
@@ -436,6 +441,10 @@ export function transformClassLikeDeclaration(state: TransformState, node: ts.Cl
 		luau.list.pushList(statementsInner, transformPropertyDeclaration(state, property, internalName));
 	}
 
+	for (const staticBlock of staticBlockDeclarations) {
+		luau.list.pushList(statementsInner, transformBlock(state, staticBlock.body));
+	}
```

Comment:
Static blocks can be interwoven with static declarations and should all run in the order they're declared:

```ts
class X {
    static value1 = 1;
    static {
        X.value1++;
    }
    static value2 = X.value1;
}

console.log(X.value1, X.value2); // 2, 2
```

[TypeScript playground](https://www.typescriptlang.org/play?target=99#code/MYGwhgzhAEAa0G8BQ1XQgFzBglsaAbmCAK4CmAjNALzQUDcKam2eiTaasAdEaZQGoBjTgF8O6LLnx9yAJhpxexcgyTikwAPYA7CFpBluILQHMAFD1mUANEutyAlPWgB6V9Dl25SIA)

So we'll need to mix them with the `staticProperties` array.

Reply:
Slightly better example using strings (to show ordering)

```ts
class X {
    static value1 = "a";
    static {
        X.value1 += "b";
    }
    static value2 = X.value1 + "c";
}

console.log(X.value1, X.value2); // ab, abc
```

[TypeScript playground](https://www.typescriptlang.org/play?target=99#code/MYGwhgzhAEAa0G8BQ1XQgFzBglsaAbmCAK4CmAjNALzQBEYdA3Cmptnoq2mrAHRFSlaAGpadAEbNu0AL4z2ufIPIAmGnAHFyVEfWDT5SYAHsAdhBMgyfECYDmACn4rKAGk2vVASibQA9P7QYBIeIcBIQA)

We should make a unit test for this